client/python/lib/test/test_tensors.py

```diff
@@ -75,32 +75,258 @@ def test_make_ndarray_valid_no_content_string_with_other_type_content():
 def test_make_ndarray_invalid_type():
     tensor_proto = TensorProto(tensor_shape=shape)
     tensor_proto.dtype = 0
-    with pytest.raises(ValueError) as exception_info:
+    with pytest.raises(TypeError) as exception_info:
         make_ndarray(tensor_proto)
-
-    exception = exception_info.value
-    assert str(exception) == "Tensor data type invalid"
+        exception = exception_info.value
+        assert str(exception) == "Unsupported tensor type: 0"
 
 def test_make_ndarray_invalid_no_shape():
     tensor_proto = TensorProto(dtype = DataType.DT_INT8, tensor_content = bytes([1,2,3]))
     with pytest.raises(ValueError) as exception_info:
         make_ndarray(tensor_proto)
-
-    exception = exception_info.value
-    assert str(exception) == "cannot reshape array of size 3 into shape ()"
+        exception = exception_info.value
+        assert str(exception) == "cannot reshape array of size 3 into shape ()"
 
 def test_make_ndarray_invalid_shape_does_not_match():
     tensor_proto = TensorProto(tensor_shape=shape, dtype = DataType.DT_INT8, tensor_content = bytes([1,2,3,4]))
     with pytest.raises(ValueError) as exception_info:
         make_ndarray(tensor_proto)
+        exception = exception_info.value
+        assert str(exception) == "cannot reshape array of size 4 into shape (3,)"
+
+# def test_make_ndarray_invalid_no_type():
+#     tensor_proto = TensorProto(tensor_shape=shape, tensor_content = bytes([1,2,3]))
+#     with pytest.raises(ValueError) as exception_info:
+#         make_ndarray(tensor_proto)
+#         exception = exception_info.value
+#         assert str(exception) == "Tensor data type invalid"
+
+def test_make_tensor_proto_valid_int():
```

Comment:
Can we parametrize this test or run make_tensor_proto multiple times? I'd like to see calls like:
```
make_tensor_proto(values)
make_tensor_proto(values, dtype)
make_tensor_proto(values, shape)
make_tensor_proto(values, dtype, shape)
```
With the same `values` + `dtype` and/or `shape` set to make all this calls equivalent (produce exactly the same proto). Such thing could be applied to all cases that does not require non-default `shape` and `dtype` configuration